### PR TITLE
DDF-1681 Fixed CSW Source Id Refresh

### PR DESCRIPTION
**Pull request by bdeining**
@wmcnalli is hero
@millerw8 
@shane-voisard

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/codice/ddf/pull/360" target="_blank">ReviewNinja

File: catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java [hunk 1]
```diff
@@ -384,7 +384,7 @@ public void refresh(Map<String, Object> configuration) throws SecurityServiceExc
 
         String idProp = (String) configuration.get(ID_PROPERTY);
         if (StringUtils.isNotBlank(idProp)) {
-            cswSourceConfiguration.setId(idProp);
+            setId(idProp);
```
Comment (stustison) [inline catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java]: So why did this one line change, but all the rest of the properties are still getting set in the cswSourceConfiguration object? It also looks like cswSourceConfiguration.getId() is called later on in this method, which would now presumably return the incorrect value. Something doesn't seem right about this change.

Comment (bdeining) [inline catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java]: The method setId() sets the cswSourceConfiguration ID and sets the ID in super.  The subsequent cswSourceConfiguration.getId() calls should return the right value.

Comment (wmcnalli) [inline catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java]: The id change looks fine, but it doesn't look like refresh() takes in account a changed url.  The SecureCxfClientFactory needs to be reconstructed with a new url if changed in the config.  I'm fine fixing it as part of this ticket  !fix

Comment (bdeining) [inline catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java]: !fixed

**Comment by wmcnalli** [comment]
:+1:

**Pull request merged**